jdaviz/configs/specviz/plugins/viewers.py

```diff
@@ -361,8 +361,10 @@ def add_data(self, data, color=None, alpha=None, **layer_state):
         result = super().add_data(data, color, alpha, **layer_state)
 
         if reset_plot_axes:
-            self.state.x_display_unit = data.get_component(self.state.x_att.label).units
-            self.state.y_display_unit = data.get_component("flux").units
+            x_units = data.get_component(self.state.x_att.label).units
+            y_units = data.get_component("flux").units
+            self.state.x_display_unit = x_units if len(x_units) else None
+            self.state.y_display_unit = y_units if len(y_units) else None
```

Comment:
I don't think len is needed.
```suggestion
            self.state.x_display_unit = x_units if x_units else None
            self.state.y_display_unit = y_units if y_units else None
```

Reply:
woops, turns out it was necessary (CI fails everywhere otherwise)

Reply:
Hmm, weird, but ok.